v2/spiffetls/spiffetls_test.go

```diff
@@ -472,3 +456,48 @@ func setWorkloadAPIResponse(ca *test.CA, s *fakeworkloadapi.WorkloadAPI, spiffeI
 		SVIDs:  []*x509svid.SVID{ca.CreateX509SVID(spiffeID)},
 	})
 }
+
+func setupTestEnv(t *testing.T) (*testEnv, func()) {
```

Comment:
Moving this to function has the side effect of requiring more code to do proper cleanup if any of the setup below fails. For example, the wlCancel callback, and workload api clients (that aren't being closed right now in the original code!!) and the sources all need to be cleaned up if any of the `require` assertions fail below.

Reply:
Good catch!
I'll fix the `setupTestEnv` function to do the proper cleanup. Hopefully moving this to a function makes the code easier to follow and not harder. I'm happy to change this if this is not really helping.

Reply:
I'd rather keep the `func()` returned by `setupTestEnv` as it makes it harder for somebody to accidentally not invoke to do the cleanup. The `func()` returned by `setupTestEnv` can invoke the cleanup method.